core-plugins/src/main/java/co/cask/hydrator/plugin/realtime/sink/RealtimeKinesisStreamSink.java

```diff
@@ -0,0 +1,217 @@
+import co.cask.hydrator.plugin.common.Properties;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.PutRecordRequest;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+
+
+
+/**
+ * A {@link RealtimeSink} that writes data to a Amazon Kinesis stream.
+ * If Kinesis Stream does not exist, it will be created using properties provided with this sink.
+ */
+@Plugin(type = RealtimeSink.PLUGIN_TYPE)
+@Name("KinesisSink")
+@Description("Real-time sink that outputs to a specified AWS Kinesis stream.")
+public class RealtimeKinesisStreamSink extends RealtimeSink<StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RealtimeKinesisStreamSink.class);
+  private final KinesisConfig config;
+  private AmazonKinesisClient kinesisClient;
+
+  public RealtimeKinesisStreamSink(KinesisConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.name),
+                                "Stream name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessKey),
+                                "Access Key should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessSecret),
+                                "Access Key secret should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.partition),
+                                "Partition name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.bodyField),
+                                "Input field should be non-null, non-empty.");
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    validateInputSchema(inputSchema);
+  }
+
+  private void validateInputSchema(@Nullable Schema inputSchema) {
+    if (inputSchema == null) {
+      return;
+    }
+    // Check the existence of field in input schema
+    Schema.Field inputSchemaField = inputSchema.getField(config.bodyField);
+    if (inputSchemaField == null) {
+      throw new IllegalArgumentException(
+        "Field " + config.bodyField + " is not present in the input schema");
+    }
+
+    // Check that the field is of primitive type
+    Schema fieldSchema = inputSchemaField.getSchema();
+    Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+    if (!fieldType.isSimpleType()) {
+      throw new IllegalArgumentException(String.format(
+        "Type %s for field %s is not supported. Supported types are Boolean, INT, LONG, FLOAT, DOUBLE, STRING, Bytes",
```

Comment:
why are Boolean and Bytes not all caps? Infact, why not just use the enum itself?